### PR TITLE
POC CRUD operations for efs fileystems and mount targets

### PR DESCRIPTION
**Pull request by fishnix**
This is a proof of concept version of the CRUD operations for a filesystem - it operates on an efs filesystem and mount targets currently.  Unfortunately, the operations in EFS are very asynchronous and we will need to implement some kind of background job function to make this work. (or we fire-and-forget from the frontend for deletes 😆 )

### CODE REVIEW

**Comment by damnski** [comment]
:+1:

File: api/types.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	GitHash string `json:"githash"`
+	// The build timestamp of the API
+	BuildStamp string `json:"buildstamp"`
+}
+
+// listFileSystemsResponse is the response for a list filesystems request
+type listFileSystemsResponse []string
+
+// FileSystemCreateRequest is the request input for creating a filesystem
+type FileSystemCreateRequest struct {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply api/types.go:26]: good point 👍

File: api/types.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+	// KMSKeyId used to encrypt the filesystem
+	KmsKeyId string
+	// Tags to apply to the filesystem
+	Tags []*Tag
+}
+
+// FileSystemResponse represents a full filesystem service response
+//
+// A filesystem can have zero or more mount targets and zero or more access points.
+type FileSystemResponse struct {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply api/types.go:38]: I'm not sure we decided to do that - I wanted to explore it here since it's greenfield.

**Comment by tenyo** [comment]
👍

**Pull request merged**